src/lazy/expanded/macro_table.rs

```diff
@@ -183,9 +182,9 @@ impl MacroTable {
     // is expected to change as development continues. It is currently used in several unit tests.
     pub const FIRST_USER_MACRO_ID: usize = Self::NUM_SYSTEM_MACROS;
 
-    pub fn new() -> Self {
-        let macros_by_id = vec![
-            Macro::named(
+    fn system_macros() -> Vec<Rc<Macro>> {
+        vec![
+            Rc::new(Macro::named(
```

Comment:
🗺️ I intend to make several thread locals that can be cheaply referenced during encoding directive construction. For now though, I build the system module for each new encoding directive.